src/OpenLoco/Map/Tile.h

```diff
@@ -398,6 +404,9 @@ namespace OpenLoco::Map
         uint8_t _5;
         uint8_t _6;
         uint8_t _7;
+
+    public:
+        uint8_t signalObjectId() const { return _5 >> 4; } // _5u
```

Comment:
I'm not sure where it is stored yet, but this is incorrect.

Reply:
you can probably work it out backwards. Look in the original for users of the signal object and then find all the users.

Reply:
lower nibble of _6 looks to be what you want.

Reply:
Hmm, it's not absolutely clear to me either. My patch:
```diff
diff --git a/src/OpenLoco/Map/Tile.h b/src/OpenLoco/Map/Tile.h
index 10508c2b..62b82c33 100644
--- a/src/OpenLoco/Map/Tile.h
+++ b/src/OpenLoco/Map/Tile.h
@@ -412,7 +412,8 @@ namespace OpenLoco::Map
         uint8_t _7;
 
     public:
-        uint8_t signalObjectId() const { return _5 >> 4; } // _5u
+        uint8_t signalObjectId() const { return _4 & 0xF; } // _4l
+        uint8_t signalObjectId2() const { return _6 & 0xF; } // _6l
     };
 
     struct road_element : public tile_element_base
```

When testing, both _4l and _6l appear to contain 0 or 1, but they do not appear to consistently correspond to either the colourlight or semaphore objects I'm selecting on the map. :confused:

Reply:
Perhaps one of these variables indicates what side of the track the signal is on.

![Screenshot_20210321_172438](https://user-images.githubusercontent.com/604665/111912532-5dcfd480-8a6a-11eb-8b1b-1b29e2d01569.png)

Reply:
Well, luckily we have a tile inspector to look at the actual bytes… At first I thought it might be 4u, but that doesn't appear to be consistent. Here's a few examples from the same map:

Semaphores:
![Screenshot (1)](https://user-images.githubusercontent.com/604665/111914250-90c99680-8a71-11eb-830a-031fd0725f80.png)
![Screenshot (2)](https://user-images.githubusercontent.com/604665/111914253-91622d00-8a71-11eb-9dee-4a2c0ceaf440.png)


Colourlights:
![Screenshot (3)](https://user-images.githubusercontent.com/604665/111914257-96bf7780-8a71-11eb-9266-6a594431b371.png)
![Screenshot (4)](https://user-images.githubusercontent.com/604665/111914258-97580e00-8a71-11eb-8ff6-ba3e98b7ef7d.png)
![Screenshot (5)](https://user-images.githubusercontent.com/604665/111914259-97580e00-8a71-11eb-9f26-385999da80b0.png)

Reply:
Confusingly, I'm not sure the tile element data is showing up correctly… I'd expect the first byte (type) to be consistent throughout signal elements at least. That is not the case.

Reply:
Well from my reading of users of signals objects it's definitely accessing the 4th or 6th but it's possible that it's not actually from the signal element. As we haven't implemented these functions I'm taking guesses as to the element.